### PR TITLE
[Core] Refactor RectifiedAnalyzer: remove exclude _Class

### PR DESCRIPTION
**Pull request by samsonasik**
This is next step of https://github.com/rectorphp/rector-src/pull/1681 for verify node re-changed by rule for : 

```bash
Same Rector Rule <-> Same Node <-> Same File
```

which previously still excluded the `Class_`Node.

This PR remove `_Class` from exclusion with check:

- applied check `AttributeKey::CREATED_BY_RULE` in both Node and original Node.
- compare original node name and changed node name if they are a string
- check if `AttributeKey::CREATED_BY_RULE` is not empty, and the applied rule is not part of it, it means the changes are in deep body, which repetitive apply with same Rector Rule -> Same Node -> same File may happen.

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
I am merging it ;)

**Comment by TomasVotruba** [comment]
Shoud be fine. We'll need some test for this mechanism, as it is getting complicated and hard to follow.
Could you add one?

**Comment by samsonasik** [comment]
The tests for removing `_Class` check without the original node name check already in tests, just remove only instance of check will cause error in tests/Issues

**Comment by TomasVotruba** [comment]
That's a good start. I mean standalone test for something like `RectorChangesCollector`. So we have it isolated from `AbstractRector` class. Now it's getting quite full as well, while it should be standalone responsibility.

**Comment by samsonasik** [comment]
I will try.
